packages/opentelemetry-types/src/trace/instrumentation/Plugin.ts

```diff
@@ -61,23 +63,18 @@ export interface PluginConfig {
   path?: string;
 
   /**
-   * Request methods that match any string in ignoreMethods will not be traced.
+   * These plugin options override the values provided in the
+   * shared plugin options section.
    */
-  ignoreMethods?: string[];
-
-  /**
-   * URLs that partially match any regex in ignoreUrls will not be traced.
-   * In addition, URLs that are _exact matches_ of strings in ignoreUrls will
-   * also not be traced.
-   */
-  ignoreUrls?: Array<string | RegExp>;
+  options?: PluginOptions;
+}
 
-  /**
-   * List of internal files that need patch and are not exported by
-   * default.
-   */
-  internalFilesExports?: PluginInternalFiles;
+export type IgnoreMatcher = string | RegExp | ((url: string) => boolean);
 
+/**
```

Comment:
I think mixing plugin options between web and node is very bad idea. Things like `database`, `dns` etc will never be shared on web, and `xhr` will never be shared on node.
I think that such changes w discuss could discuss on sig meeting and how we want to handle that. I would rather have web and node separation like this is done for example in `core/platform`.

Reply:
We can talk about it in SIG but I thought this was already pretty much agreed. We've had a couple discussions about it already.